### PR TITLE
Add Door Lock: Weaknesses

### PR DESCRIPTION
**Pull request by henriquegemignani**
This swaps an entire weakness at once with another one, allowing this generation to be fast and compatible with multiworld.

### CODE REVIEW

File: randovania/generator/dock_weakness_distributor.py [hunk 1]
```diff
@@ -45,18 +45,45 @@ def distribute_pre_fill_weaknesses(patches: GamePatches):
...
+                (node, weakness_database.dock_rando_params[node.dock_type].unlocked)
+                for node in dock_nodes
+                if (
+                        node not in unlocked
+                        and game.world_list.node_by_identifier(node.default_connection) in unlocked
+                )
+            ])
+        patches = patches.assign_weaknesses_to_shuffle([(node, True) for node, _ in docks_to_unlock])
+
+    else:
```
Comment (duncathan) [inline randovania/generator/dock_weakness_distributor.py:65]: should probably be an else if

File: randovania/gui/ui_files/preset_dock_rando.ui [hunk 2]
```diff
@@ -103,7 +92,7 @@
            <item row="5" column="0" colspan="2">
             <widget class="QLabel" name="label">
              <property name="text">
-              <string>Only Vanilla is compatible with multiworld sessions.</string>
+              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The full door lock mode is &lt;span style=&quot; font-weight:700;&quot;&gt;not&lt;/span&gt; compatible with multiworld sessions.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Comment (duncathan) [inline randovania/gui/ui_files/preset_dock_rando.ui:95]: this is both the opposite of the truth, and is using the old name lol

File: randovania/layout/base/dock_rando_configuration.py [hunk 3]
```diff
@@ -200,6 +202,6 @@ def is_enabled(self) -> bool:
 
     def settings_incompatible_with_multiworld(self) -> list[str]:
         danger = []
-        if self.is_enabled():
-            danger.append("Door Lock randomizer")
+        if self.mode == DockRandoMode.DOCKS:
+            danger.append("Full Door Lock randomizer")
```
Comment (duncathan) [inline randovania/layout/base/dock_rando_configuration.py:206]: old name

File: randovania/layout/preset_migration.py [hunk 4]
```diff
@@ -802,12 +807,21 @@ def _migrate_v49(preset: dict) -> dict:
...
     return preset
 
+
 def _migrate_v50(preset: dict) -> dict:
     if preset["game"] == "dread":
         preset["configuration"]["raven_beak_damage_table_handling"] = "consistent_low"
     return preset
 
 
+def _migrate_v51(preset: dict) -> dict:
```
Comment (duncathan) [inline randovania/layout/preset_migration.py:817]: ...did we add a migration to change one-way in the preset?

**Review by duncathan — CHANGES_REQUESTED** [review]

File: randovania/gui/ui_files/preset_dock_rando.ui (hunk 2 above)
Comment (henriquegemignani) [reply randovania/gui/ui_files/preset_dock_rando.ui:95]: there's a bolded "not" in the escaped html mess there.

File: randovania/layout/preset_migration.py (hunk 4 above)
Comment (henriquegemignani) [reply randovania/layout/preset_migration.py:817]: I was ignoring with the lack of use, but eh could use it

File: randovania/gui/ui_files/preset_dock_rando.ui [hunk 5]
```diff
@@ -101,9 +90,9 @@
             </widget>
            </item>
            <item row="5" column="0" colspan="2">
-            <widget class="QLabel" name="label">
+            <widget class="QLabel" name="multiworld_label">
              <property name="text">
-              <string>Only Vanilla is compatible with multiworld sessions.</string>
+              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The selected type is &lt;span style=&quot; font-weight:700;&quot;&gt;not&lt;/span&gt; compatible with multiworld sessions.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Comment (duncathan) [inline randovania/gui/ui_files/preset_dock_rando.ui:95]: do you mean *in*compatible?

**Review by duncathan — APPROVED** [review]

File: randovania/gui/ui_files/preset_dock_rando.ui (hunk 2 above)
Comment (duncathan) [reply randovania/gui/ui_files/preset_dock_rando.ui]: this is why I've been slowly changing every label I touch to markdown lmao

**Pull request merged**